### PR TITLE
[Search] Update backend system guide

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

Add examples to the migration guide for Search backend systems.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/features/search/how-to-guides.md [hunk 1]
```diff
@@ -386,14 +386,27 @@ import { searchModuleTechDocsCollator } from '@backstage/plugin-search-backend-m
...
-// extending search with collator modules to start index documents, take in optional schedule parameters.
+// [3] extending search with collator modules to start index documents, take in optional schedule parameters.
 backend.add(searchModuleCatalogCollator());
 backend.add(searchModuleTechDocsCollator());
 backend.add(searchModuleExploreCollator());
 
 backend.start();
 ```
+
+To create your own collators/decorators modules, please use the [searchModuleCatalogCollator](https://github.com/backstage/backstage/blob/d7f955f300893f50c4882ea8f5c09aa42dfaacfd/plugins/search-backend-module-catalog/src/alpha.ts#L49) as an example, we recommend that modules have their own packages. You can also find the the available search engines and indexers modules documentation in the Alpha API reports:
```
Comment (emmaindal) [inline docs/features/search/how-to-guides.md:401]: > we recommend that modules have their own packages.

I was thinking for example decorators and collators for catalog can live in the same package, do you agree?

**Review by emmaindal — COMMENTED** [review]
💯 just one small comment

File: docs/features/search/how-to-guides.md [hunk 2]
```diff
@@ -386,14 +386,27 @@ import { searchModuleTechDocsCollator } from '@backstage/plugin-search-backend-m
...
+
+**Search engines**
+
+- Postgres [module](https://github.com/backstage/backstage/blob/d7f955f300893f50c4882ea8f5c09aa42dfaacfd/plugins/search-backend-module-pg/alpha-api-report.md);
+- Elasticsearch [module](https://github.com/backstage/backstage/blob/d7f955f300893f50c4882ea8f5c09aa42dfaacfd/plugins/search-backend-module-elasticsearch/alpha-api-report.md).
+
+**Search indexers**
```
Comment (emmaindal) [inline docs/features/search/how-to-guides.md:408]: Maybe we can use "Search collators/decorators" to follow our same terminology as explained in the concepts documentation?

File: docs/features/search/how-to-guides.md (hunk 1 above)
Comment (camilaibs) [reply docs/features/search/how-to-guides.md]: Done!

File: docs/features/search/how-to-guides.md (hunk 2 above)
Comment (camilaibs) [reply docs/features/search/how-to-guides.md]: Done!

**Review by emmaindal — APPROVED** [review]

**Pull request merged**